openfisca_uk/variables/finance/benefit/family/universal_credit.py

```diff
@@ -1,134 +1,359 @@
-            benunit.members("childcare_cost", period)
+        return UC.elements.carer.amount * has_carer * MONTHS_IN_YEAR
+
+
+class UC_housing_costs_element(Variable):
+    value_type = float
+    entity = BenUnit
+    label = u"UC housing costs element"
+    definition_period = YEAR
+
+    def formula(benunit, period, parameters):
+        tenure_type = benunit.value_from_first_person(
+            benunit.members.household("tenure_type", period).decode_to_str()
         )
-        childcare_element = min_(
-            childcare_limit,
-            eligible_childcare_costs * UC.elements.childcare_cost_rate,
+        max_housing_costs = select(
+            [
+                (tenure_type == TenureType.RENT_FROM_COUNCIL.name)
+                | (tenure_type == TenureType.RENT_FROM_HA.name),
+                tenure_type == TenureType.RENT_PRIVATELY.name,
+                True,
+            ],
+            [
+                benunit("benunit_rent", period),
+                min_(
+                    benunit("LHA_cap", period), benunit("benunit_rent", period)
+                ),
+                0,
+            ],
+        )
+        return max_housing_costs - benunit("UC_non_dep_deductions", period)
+
+
+class UC_individual_non_dep_deduction(Variable):
+    value_type = float
+    entity = Person
+    label = u"Non-dependent deduction (individual)"
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
+        not_rent_liable = person.benunit("benunit_rent", period) == 0
+        over_21 = person("age", period) >= 21
+        deduction = parameters(
+            period
+        ).benefit.universal_credit.elements.housing.non_dep_deduction
+        return deduction * over_21 * not_rent_liable * MONTHS_IN_YEAR
+
+
+class UC_non_dep_deductions(Variable):
+    value_type = float
+    entity = BenUnit
+    label = u"Non-dependent deductions"
+    definition_period = YEAR
+
+    def formula(benunit, period, parameters):
+        non_dep_deductions_in_hh = benunit.max(
+            benunit.members.household.sum(
+                benunit.members("UC_individual_non_dep_deduction", period)
+            )
+        )
+        non_dep_deductions_in_bu = benunit.sum(
+            benunit.members("UC_individual_non_dep_deduction", period)
+        )
+        return non_dep_deductions_in_hh - non_dep_deductions_in_bu
+
+
+class UC_individual_disabled_element(Variable):
+    value_type = float
+    entity = Person
+    label = u"Disabled element of UC"
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
+        UC = parameters(period).benefit.universal_credit
+        return person("is_disabled_for_benefits", period) * where(
+            person("is_child", period),
+            UC.elements.child.disabled.amount,
+            UC.elements.disabled.amount,
         )
-        return premiums + childcare_element
 
 
-class UC_eligible_rent(Variable):
+class UC_individual_severely_disabled_element(Variable):
```

Comment:
could you note in a comment whether non-severely disabled people also get this? i.e. does `is_severely_disabled_for_benefits` imply `is_disabled_for_benefits`?

Reply:
Added to the documentation field - should use that more